README.md

```diff
@@ -1,2 +1,158 @@
-# helm-mirror
+# Helm Mirror Plugin
+
 Helm plugin used to mirror repositories
+
+## Usage
+
+```shell
+Mirror Helm Charts from an index file into a local folder.
+
+For example:
+
+helm-mirror https://yourorg.com/charts /yourorg/charts
+
+This will download the index file and the charts into
+the folder indicated.
+
+The index file is a yaml that contains a list of
+charts in this format. Example:
+
+        apiVersion: v1
+        entries:
+          chart:
+          - apiVersion: 1.0.0
+            created: 2018-08-08T00:00:00.00000000Z
+            description: A Helm chart for your application
+            digest: 3aa68d6cb66c14c1fcffc6dc6d0ad8a65b90b90c10f9f04125dc6fcaf8ef1b20
+            name: chart
+            urls:
+            - https://kubernetes-charts.yourorganization.com/chart-1.0.0.tgz
+          chart2:
+          - apiVersion: 1.0.0
+            created: 2018-08-08T00:00:00.00000000Z
+            description: A Helm chart for your application
+            digest: 7ae62d60b61c14c1fcffc6dc670e72e62b91b91c10f9f04125dc67cef2ef0b21
+            name: chart
+            urls:
+            - https://kubernetes-charts.yourorganization.com/chart2-1.0.0.tgz
+
+This will download these charts:
+
+        https://kubernetes-charts.yourorganization.com/chart-1.0.0.tgz
+        https://kubernetes-charts.yourorganization.com/chart2-1.0.0.tgz
+
+into your destination folder.
+
+Usage:
+  helm-mirror [Repo URL] [Destination Folder] [flags]
+  helm-mirror [command]
+
+Available Commands:
+  help           Help about any command
+  inspect-images Extract all the images of the Helm Charts.
+  version        Show version of the helm mirror plugin
+
+Flags:
+  -h, --help      help for helm-mirror
+  -v, --verbose   verbose output
+
+Use "helm-mirror [command] --help" for more information about a command.
+```
+
+## Commands
+
+### inspect-images
+
+```shell
+./mirror inspect-images -h
+Extract all the images of the Helm Chart or
+the Helm Charts in the folder provided. This command dumps
+the images on 'stdout' by default, for more options check
```

Comment:
It might be worth fetching the container image/versions and pushing them to an alternative location. As an opt in flag? Just thinking out loug.

Reply:
We are working to address [these](https://flavio.castelli.me/2018/07/18/hackweek-project-docker-registry-mirror/) uses cases. So this tool is just a piece of the biggest puzzle :)

We already have a [patch for skopeo](https://github.com/containers/skopeo/pull/524) that provides an easy way to sync images between registries. The idea is, once this PR gets merged, to provide a new output format that would be the the `yaml` configuration file used by `skopeo sync`.

The format of this configuration file is described over [there](https://github.com/containers/skopeo/pull/524/files#skopeo-sync-1).